WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -1738,12 +1737,21 @@ private extension EditableOrderViewModel {
         orderSynchronizer.removeCoupon.send(code)
     }
 
-    func addFee(_ formattedFeeLine: String) {
-        let feeLine = OrderFactory.newOrderFee(total: formattedFeeLine)
-        orderSynchronizer.setFee.send(feeLine)
+    func addFee(with total: String, name: String? = nil) {
+        let feeLine = OrderFactory.newOrderFee(total: total, name: name)
+        orderSynchronizer.addFee.send(feeLine)
```

Comment:
I believe this works for our case but introduces a bug in the previous implementation, since only one fee is allowed per order, but here we're changing the internals of the method to `add` fees rather than `set` them. In the `OrderSynchronizer` this means we're appending the fees:

```
addFee.withLatestFrom(orderPublisher)
  ...
  let updatedOrder = FeesInputTransformer.append(input: feeLineInput, on: order)
  // Calculate order total locally while order is being synced
  return OrderTotalsCalculator(for: updatedOrder, using: self.currencyFormatter).updateOrderTotal()
}
```

This causes the fee section to become a sort of stack of custom amount values, where these can be accumulated and are edited/removed from the stack one by one (but only the last one is visible when editing)
![Simulator Screen Recording - iPhone 15 - 2023-10-24 at 09 37 30](https://github.com/woocommerce/woocommerce-ios/assets/3812076/73a864af-a2ad-4a43-a0e8-737b481d1334)

Perhaps the simplest approach is to just duplicate the method as `addCustomAmount()`, and leave the previous `addFee()` as it is? (We may need to update the order sync as well). Or add an extra check for which method to call based on feature flag? It may also help when removing fee-related code after the feature is released.

Along with this would be worth to add some unit tests checking if fee per order is unique when the feature flag is switched off?

Reply:
Thanks for the comment @iamgabrielma 

Actually, we are not using `addFee` for the previous implementation, but `setFee`, which keeps the same logic as before. 
If we test the previous implementation we have to do it without the custom amounts, as they are totally independent and won't be in the app at the same time. You can do that by disabling the `orderCustomAmountsM1` feature flag.

Reply:
Perfect, thanks for the explanation. The "bug" I was seeing was because I was adding fees through custom amounts, which due to the flag merchants won't be able to do 👍